packages/arb-token-bridge-ui/src/components/TransferPanel/TransferPanel.tsx

```diff
@@ -207,16 +261,34 @@ export function TransferPanel() {
   }, [connectionState, importTokenModalStatus])
 
   useEffect(() => {
-    const getDestinationAddressType = async () => {
-      setIsDestinationAddressSmartContract(
-        await addressIsSmartContract(
-          String(destinationAddress),
-          isDepositMode ? l2Provider : l1Provider
-        )
-      )
+    let isLatestUpdate = true
+    const validateTransfer = async () => {
+      const funcProps = {
+        from: walletAddress,
+        to: destinationAddress,
+        l1Provider,
+        l2Provider,
+        isDeposit: isDepositMode
+      }
+      const error = await (isSmartContractWallet
+        ? getSmartContractTransferError(funcProps)
+        : getEOATransferError(funcProps))
+      if (isLatestUpdate) {
```

Comment:
Didn't understand the reason behind using `isLatestUpdate` 🤔

Reply:
We do an async check on destination address input change. We want to unsubscribe on a new keystroke so that we don't update the state with old values (and cause side effects)